public/app/features/trails/MetricSelect/util.ts

```diff
@@ -0,0 +1,51 @@
+// Consider any sequence of characters not permitted for metric names as a sepratator
+const splitSeparator = /[^a-z0-9_:]+/;
+
+export function deriveSearchTermsFromInput(whiteSpaceSeparatedTerms?: string) {
+  return (
+    whiteSpaceSeparatedTerms
+      ?.toLowerCase()
+      .split(splitSeparator)
+      .filter((term) => term.length > 0) || []
+  );
+}
+
+export function createJSRegExpFromSearchTerms(searchTerms: string[]) {
```

Comment:
what don't get is why this searchTerms is not simply the plain searchQuery string, you can do the split here and build the regex from the terms, same inside createPromRegExp. 

Then no need for the custom variable, custom URL sync, etc. 

(technically no need for variable, just a state prop that is URL synced)

Reply:
>what don't get is why this searchTerms is not simply the plain searchQuery string

No longer any reason. I am going to change it to be so.

>then no need for the custom variable

Hmm, yes...  Here is how it's used currently.
```ts
    const match = sceneGraph.interpolate(trail, '{${filters}${metricSearch:filter}}');
```
Arguably, I could construct this query parameter without `metricSearch` being an interpolated variable.

>custom URL sync, etc.
Yeah, that's already gone. Much less to worry about as full string.

> But a variable could be easier as you get the URL sync for free
We could have `searchQuery` be a state prop on DataTrail, to ensure that it persists when metric select scene gets wiped. It shouldn't be hard to urlSync this alongside `metric` in DataTrail. Hoping it's just a matter of appending the key to the list.